public/app/core/utils/UserProvider.tsx

```diff
@@ -15,36 +20,70 @@ export interface ChangePasswordFields {
   confirmNew: string;
 }
 
+export interface ProfileUpdateFields {
+  name: string;
+  email: string;
+  login: string;
+}
+
 export interface Props {
-  children: (api: UserAPI, states: LoadingStates) => JSX.Element;
+  loadUser?: boolean; // if true, will load user on mount
+  children: (api: UserAPI, states: LoadingStates, user: User) => JSX.Element;
 }
 
 export interface State {
+  user: User;
   loadingStates: LoadingStates;
 }
 
 export class UserProvider extends PureComponent<Props, State> {
   state: State = {
+    user: {} as User,
     loadingStates: {
       changePassword: false,
+      loadUser: true,
+      updateUserProfile: false,
     },
   };
 
+  componentDidMount() {
+    if (this.props.loadUser) {
+      this.loadUser();
+    }
+  }
+
   changePassword = async (payload: ChangePasswordFields) => {
     this.setState({ loadingStates: { ...this.state.loadingStates, changePassword: true } });
     await getBackendSrv().put('/api/user/password', payload);
     this.setState({ loadingStates: { ...this.state.loadingStates, changePassword: false } });
   };
 
+  loadUser = async () => {
+    this.setState({
+      loadingStates: { ...this.state.loadingStates, loadUser: Object.keys(this.state.user).length === 0 },
+    });
+    const user = await getBackendSrv().get('/api/user');
+    this.setState({ user, loadingStates: { ...this.state.loadingStates, loadUser: Object.keys(user).length === 0 } });
+  };
+
+  updateUserProfile = async (payload: ProfileUpdateFields) => {
+    this.setState({ loadingStates: { ...this.state.loadingStates, updateUserProfile: true } });
+    await getBackendSrv().put('/api/user', payload);
+    await this.loadUser();
+    this.setState({ loadingStates: { ...this.state.loadingStates, updateUserProfile: false } });
+  };
+
   render() {
     const { children } = this.props;
-    const { loadingStates } = this.state;
+    const { loadingStates, user } = this.state;
 
     const api = {
       changePassword: this.changePassword,
+      loadUser: this.loadUser,
+      updateUserProfile: this.updateUserProfile,
     };
 
-    return <>{children(api, loadingStates)}</>;
+    return <>{children(api, loadingStates, user)}</>;
```

Comment:
Since it's `UserProvider` I think the order should be `user, api, loadingStates`